core/src/components/header/header.utils.ts

```diff
@@ -0,0 +1,133 @@
+import { readTask, writeTask } from '@stencil/core';
+
+const TRANSITION = 'all 0.2s ease-in-out';
+
+export const cloneElement = (tagName: string) => {
+  const getCachedEl = document.querySelector(`${tagName}.ion-cloned-element`);
+  if (getCachedEl !== null) { return getCachedEl; }
+
+  const clonedEl = document.createElement(tagName);
+  clonedEl.classList.add('ion-cloned-element');
+  clonedEl.style.setProperty('display', 'none');
+  document.body.appendChild(clonedEl);
+
+  return clonedEl;
+};
+
+export const createHeaderIndex = (headerEl: any): any | undefined => {
+  if (!headerEl) { return; }
+
+  const toolbars = headerEl.querySelectorAll('ion-toolbar');
+
+  return {
+    el: headerEl,
+    toolbars: Array.from(toolbars).map((toolbar: any) => {
+      const ionTitleEl = toolbar.querySelector('ion-title');
+      return {
+        el: toolbar,
+        background: toolbar.shadowRoot!.querySelector('.toolbar-background'),
+        ionTitleEl,
+        innerTitleEl: (ionTitleEl) ? ionTitleEl.shadowRoot!.querySelector('.toolbar-title') : null,
+        ionButtonsEl: Array.from(toolbar.querySelectorAll('ion-buttons'))
+      };
+    })
+  };
+};
+
+const clampValue = (value: number, max: number, min: number): number => {
```

Comment:
i recall we have some clamp() aux function somewhere, maybe we can import from there